src/middlewared/middlewared/utils/allowlist.py

```diff
@@ -19,10 +21,10 @@ def __init__(self, allowlist):
                 self.exact.setdefault(method, set())
                 self.exact[method].add(resource)
 
-    def authorize(self, method, resource):
+    def authorize(self, method: HttpVerb, resource: NonEmptyString):
```

Comment:
Do we need to use annotated types here? Won't `str` be enough (for both)? I doubt this will work good with many IDEs and other tools

Reply:
Never _need_ to use type hints, but I think here we should. I would argue that type hints are most helpful when they're as specific as possible. IDEs/linters resolve type aliases the same way they would any other user-defined type. The default type checker in VS Code flags anything that isn't listed in the definition of `HttpVerb` if I try to annotate as being an `HttpVerb`.

Reply:
@creatorcary agreed with HttpVerb if it works, but NonEmptyString seems excessive

Reply:
Yeah that's fair.